src/scheduler/resource_resv.c

```diff
@@ -1469,14 +1469,15 @@ remove_resresv_from_array(resource_resv **resresv_arr,
  *
  * @param[in]	resresv_arr	-	job array to add job to
  * @param[in]	resresv	-	job to add to array
+ * @param[out]  index -	    Index where resresv is added in resresv_arr
  *
  * @return	array (changed from realloc)
  * @retval	NULL	: on error
  *
  */
 resource_resv **
 add_resresv_to_array(resource_resv **resresv_arr,
-	resource_resv *resresv)
+	resource_resv *resresv, int *index)
```

Comment:
How about we flip this around?  Instead of returning the index, accept a flag instead (like SET_INDEX or something like that).  When passed, add_resresv_to_array() will set resresv->resresv_ind.  You can pass NO_FLAGS other times.